src/coreclr/src/vm/syncblk.h

```diff
@@ -875,11 +878,26 @@ class InteropSyncBlockInfo
 
 private:
     // See InteropLib API for usage.
-    void* m_externalComObjectContext;
+    PTR_VOID m_externalComObjectContext;
 
-    using ManagedObjectComWrapperByIdMap = MapSHash<INT64, void*>;
     CrstExplicitInit m_managedObjectComWrapperLock;
-    NewHolder<ManagedObjectComWrapperByIdMap> m_managedObjectComWrapperMap;
+    PTR_ManagedObjectComWrapperByIdMap m_managedObjectComWrapperMap;
```

Comment:
This looks correct but if there is concern from others it could be done with minimal refactoring here. This file could add only the friend ClrDataAccess declaration and then in request.cpp you could probably snoop the field and replicate the iteration something like this:

````C++
using ManagedObjectComWrapperByIdMap = MapSHash<INT64, void*>;
PTR_InteropSyncBlockInfo pInteropInfo = ...
DPTR(NewHolder<ManagedObjectComWrapperByIdMap>) mapHolder(PTR_TO_MEMBER_ADDR(InteropSyncBlockInfo, pInteropInfo, m_managedObjectComWrapperMap));
DPTR(ManagedObjectComWrapperByIdMap) pMap((TADDR)mapHolder.Extract());

ManagedObjectComWrapperByIdMap::Iterator iter = pMap->Begin();
while (iter != pMap->End())
{
    callback(iter->Value(), additionalData);
    ++iter;
}
````

Reply:
My plan after a brief chat with Jeff tonight is to check this in as is for 6, then do some work to minimize the code diff. Which will likely include:

- Making sure sospriv.h only has real code changes and not whitespace
- Getting rid of the sentinel value as commented on above
- Doing something like you describe to avoid changing code in syncblk.h
- Modifying the DAC memory reporting code to duplicate some code instead of refactoring the DACTryGetComWrappersObjectFromCCW method

That should make it a smaller change where everything added is new code with very little change to existing runtime code